mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -137,6 +138,18 @@ class KMeansModel private[ml] (
   @Since("1.6.0")
   override def write: MLWriter = new KMeansModel.KMeansModelWriter(this)
 
+  override def hashCode(): Int =
+    this.getClass.hashCode() + uid.hashCode() + clusterCenters.map(_.hashCode()).sum
```

Comment:
Probably not super important, but it seems like we could have a better hash code function here.